### DISCUSSION TITLE
Send a notification when a block is completed

### DISCUSSION THREAD

**Discussion by simjnd** [warpdotdev/Warp#628] [Archived - Launched]
**Is your feature request related to a problem? Please describe.**
When I launch a process that is going to take a long time (e.g. `brew update`, launching containers) I would like to know when it is complete so I don't have to stare at my terminal while it's being processed

**Describe the solution you'd like**
Having a "Notify" option when right-clicking a block that is still being processed so that its completion will trigger a push notification. That way I can minimize the window and do something else and pop it back when I've received the notification that it is done.

**Additional context**
N/A

**Comment by zheng** [comment]
We're actively working on in-app notifications! It's going to send a push notification when a very long command finishes & the relevant window is inactive. There'll be a setting to have this on or off in general.

We're also going to add icons to tabs, for when you're focused on another tab as a visual indication that a command finished.

**Comment by simjnd** [comment]
Amazing, looking forward to it!

**Comment by drpayyne** [comment]
> We're also going to add icons to tabs, for when you're focused on another tab as a visual indication that a command finished.

Does this also include a 'loader' icon on a tab when a command is executing and focus shifts away from the tab?

**Comment by elviskahoro** [comment]
Discord - feature requests - 10:15:2021
https://discord.com/channels/851854972600451112/852249738002956311/898426508702523402

![image](https://user-images.githubusercontent.com/29553206/150295792-d590dff0-3e6e-43d3-bfa0-e9f11d9534c6.png)


![image](https://user-images.githubusercontent.com/29553206/150295739-78bdc272-5c98-4eb2-b414-6dc9942fd033.png)

**Comment by szgupta** [comment]
Notifications were released in `v0.2022.05.23.09.07.stable_01`!

**Reply by dmythro** [reply]
Yes, and I really like it :) Longer build/compile tasks auto-notified while you're busy with something else is ace :)

**Comment by donaldjosephsmith** [comment]
Hi, this is a great feature, but I noticed that I got a notification after a long-running command that I control-C'd myself. Is this intended? Seems superfluous.

**Comment by rossanmol** [comment]
Can we consider the option of updating the icon in dock, when a command is running? 
This can prevent lots of notifications while providing the functionality of knowing if any command is running 

(same can be implemented in individual tabs, to keep track of each tab)


<img width="73" alt="image" src="https://user-images.githubusercontent.com/18351988/230034273-1897fe4b-acdb-4360-a9b2-137412c7f4e8.png">

**Comment by dbreton333** [comment]
Would it be possible to expend push notifications to phones? I would love to be able to be away from my computer when a long block is running and get notified on my phone when it's done running such that I can manage my time efficiently. Thanks

**Reply by shinayser** [reply]
I definetly need this!

**Comment by Roy3838** [comment]
Hi! I built a free/FOSS tool that covers this exactly: 
https://github.com/Roy3838/Observer It ships a CLI wrapper, so `observe brew update` (or whatever the long process is) will ping or call you the moment it's done, no staring at the terminal required. 
Runs locally, nothing leaves your machine. Any questions feel free to reach out! :D
